cli/cli_app.go

```diff
@@ -661,6 +665,9 @@ func runMultiModuleCommand(command string, terragruntOptions *options.Terragrunt
 // complicated and not something we handle at the moment.
 func modulesNeedInit(terragruntOptions *options.TerragruntOptions) (bool, error) {
 	modulesPath := util.JoinPath(terragruntOptions.WorkingDir, ".terraform/modules")
+	if tfDataDir, ok := os.LookupEnv("TF_DATA_DIR"); ok {
+		modulesPath = util.JoinPath(tfDataDir, "modules")
```

Comment:
Seems like a lot of copy/pasted logic. Perhaps extract into a method called something like `withTfDataDir(pathParts ...string)`? If `TF_DATA_DIR` isn't set, it returns `util.JoinPath` on `pathParts`. If it is set, it joins the data dir to `pathParts`, dropping `.terraform` if it's the first arg to `pathParts`.

Reply:
Turns out every time `.terraform` is used, it comes from `TerragruntOptions`, so I have moved the logic there.